site/app/libraries/database/DatabaseQueries.php

```diff
@@ -1029,24 +1037,83 @@ public function getForumAttachments(array $post_ids, bool $all_vers = false): ar
+            "INSERT INTO courses_users (term, course, user_id, user_group, registration_section, manual_registration)
+            VALUES (?,?,?,?,?,?)",
+            $params
+        );
+
+        $params = [
+            $user->getRotatingSection(),
+            $user->getRegistrationSubsection(),
+            $user->getRegistrationType(),
+            $user->getId()
+        ];
+        $this->course_db->query(
+            "UPDATE users SET rotating_section=?, registration_subsection=?, registration_type=? WHERE user_id=?",
+            $params
+        );
+        $this->updateGradingRegistration($user->getId(), $user->getGroup(), $user->getGradingRegistrationSections());
+    }
+    else {
+        $params = [$semester, $course, $user_or_id, $user_group, $registration_section, $manual_registration];
+        $this->submitty_db->query(
+            "INSERT INTO courses_users (term, course, user_id, user_group, registration_section, manual_registration) 
+            VALUES (?,?,?,?,?,?)",
+            $params
+        );
+    }
+}
+
     /**
-     * @param User   $user
-     * @param string $semester
-     * @param string $course
+     * Update an existing user's course registration details
      */
-    public function insertCourseUser(User $user, $semester, $course) {
-        $params = [$semester, $course, $user->getId(), $user->getGroup(), $user->getRegistrationSection(),
-                        $this->submitty_db->convertBoolean($user->isManualRegistration())];
+    public function updateUser($user_id, $term, $course, $user_group, $registration_section, $manual_registration) {
+        $params = [$user_group, $registration_section, $manual_registration, $term, $course, $user_id];
         $this->submitty_db->query(
-            "
-INSERT INTO courses_users (term, course, user_id, user_group, registration_section, manual_registration)
-VALUES (?,?,?,?,?,?)",
+            "UPDATE courses_users 
+            SET user_group=?, registration_section=?, manual_registration=?
+            WHERE term=? AND course=? AND user_id=?", 
             $params
         );
+    }
 
-        $params = [$user->getRotatingSection(), $user->getRegistrationSubsection(), $user->getRegistrationType(), $user->getId()];
-        $this->course_db->query("UPDATE users SET rotating_section=?, registration_subsection=?, registration_type=? WHERE user_id=?", $params);
-        $this->updateGradingRegistration($user->getId(), $user->getGroup(), $user->getGradingRegistrationSections());
+    /**
+     * Get all instructors for a course
+     * @return array Array of instructor user_ids 
+     */
+    public function getInstructors($term, $course) {
+        $this->submitty_db->query(
+            "SELECT user_id FROM courses_users 
+            WHERE term=? AND course=? AND user_group=1",
+            [$term, $course]
+        );
+        return array_map(function ($row) {
+            return $row['user_id'];
+        }, $this->submitty_db->rows());
```

Comment:
This function isn't really needed, there is a function that already gets different type of users, including only instructors.